boot/zephyr/main.c

```diff
@@ -153,16 +153,20 @@ static void do_boot(struct boot_rsp *rsp)
     /* Get ram address for image */
     vt = (struct arm_vector_table *)(rsp->br_hdr->ih_load_addr + rsp->br_hdr->ih_hdr_size);
 #else
-    uintptr_t flash_base;
     int rc;
+    const struct flash_area *fap;
+    static uint32_t dst[2];
 
     /* Jump to flash image */
-    rc = flash_device_base(rsp->br_flash_dev_id, &flash_base);
+    rc = flash_area_open(rsp->br_flash_dev_id, &fap);
     assert(rc == 0);
 
-    vt = (struct arm_vector_table *)(flash_base +
-                                     rsp->br_image_off +
-                                     rsp->br_hdr->ih_hdr_size);
+    rc = flash_area_read(fap, rsp->br_hdr->ih_hdr_size, dst, 8);
+    assert(rc == 0);
+
```

Comment:
Should probably enter a lock up here if asserts are disabled and rc != 0

Reply:
Thank you, I have added some code. Please check if this works.